### PR TITLE
Refactor: Exclude schema pickle files from repo and build at runtime

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
Refactor: Exclude schema pickle files from repo and build at runtime

### CODE REVIEW

File: python-avd/scripts/build-schemas.py
Comment (carl-baillargeon) [inline python-avd/scripts/build-schemas.py:83]: print? Why not a logger?

**Review by carl-baillargeon — DISMISSED** [review]
LGTM if CI passes

File: python-avd/scripts/build-schemas.py
Comment (ClausHolbechArista) [reply python-avd/scripts/build-schemas.py:83]: This is just a cli script printing to pre-commit.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: maybe a docstring here?

Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: what should this function return?

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: Using values instead of items as not using id + using any
```suggestion
if any(not pickle_file.exists() for pickle_file in DEFAULT_PICKLED_SCHEMAS.values()):
  return True
```

Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: reading the rest, actually rather than doing three loops, why not just do the three checks in a single loop?

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: This assumes that `defs` are all top level - which is the case today but maybe worth adding a comment

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: could also use the `resolved_schema_store` when resolving the schema for `eos_designs` ?

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:119]: Maybe a log

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: do we need this when we are also deeocopying in the `_resolve_schema` function? Also should we not pop `$defs` in the `_resolve_schema` funciton instead?

File: .github/workflows/pull-request-management.yml
Comment (gmuloc) [inline .github/workflows/pull-request-management.yml]: Can we make this a file instead? we are tracking versions dependencies all over the place.
Also it says pydantic line 476 but not in the comment line 474 - is it on purpose?

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:119]: We don't do logging in AVD today.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: We only support defs at the top.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: I am looping over different files. And to avoid doing 3 checks on each I wanted to exit quickly for the most common case.

File: .github/workflows/pull-request-management.yml
Comment (ClausHolbechArista) [reply .github/workflows/pull-request-management.yml]: We can only install a subset of our usual requirements, and I think adding another requirements file will confuse. Here it is clear why it is needed.
I will add pydantic to the comment.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: the outer loop is for the same files line 46 and 51

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: we do today yes

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:119]: this is just failing silently - which can be hard to troubleshoot

File: .github/workflows/pull-request-management.yml
Comment (gmuloc) [reply .github/workflows/pull-request-management.yml]: but it means we need to maintain this file as well with dependencies versions

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: yes but since I replace in one, the check is not for the same file.

File: .github/workflows/pull-request-management.yml
Comment (ClausHolbechArista) [reply .github/workflows/pull-request-management.yml]: yes for this special case where we have to match whatever Ansible Galaxy decides is the current version.

Comment (ClausHolbechArista) [reply .github/workflows/pull-request-management.yml]: refactored

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: I wanted to avoid running the hash operation unless strictly needed, but I have combined the last two to make you happy :)

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:119]: It does not fail as such. We will just not save a pickled file, so next time we run, the schema will be resolved again.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: done

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py]: fixed

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:119]: Understood - do you think it could be puzzling in some troubleshooting situations maybe? Like seeing the pickle file not being updated silently because force_rebuild is set to True but permission is failing and this is just eaten by the function?

Not willing to block the PR for this but wondering if it could lead to head banging troubleshooting sessions :)

**Review by gmuloc — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:119]: force rebuild is only used by the pre-commit / schema tooling scripts, so not something part of AVD. We have to move on and there is no alternative for this available today.

**Pull request merged**
